onnxruntime/test/ir/graph_test.cc

```diff
@@ -115,13 +115,243 @@ static bool RegisterCustomSchemas() {
 
   return true;
 }
+static std::once_flag once;
 
-static const bool kSchemasRegistered = RegisterCustomSchemas();
+class GraphTest : public ::testing::Test {
+ protected:
+  GraphTest() {
+    std::call_once(once, RegisterCustomSchemas);
+    logger_ = DefaultLoggingManager().CreateLogger("GraphTest");
+  }
+
+  std::unique_ptr<logging::Logger> logger_;
+};
+
+static void SetTypeAndShape(TypeProto_Tensor* t, int type, const std::vector<int64_t>& shape) {
+  t->set_elem_type(type);
+  for (int64_t i : shape) {
+    TensorShapeProto* s = t->mutable_shape();
+    s->add_dim()->set_dim_value(i);
+  }
+}
+
+static void ImportOpset(ModelProto& m, const char* domain, int64_t version) {
+  OperatorSetIdProto* p = m.add_opset_import();
+  p->set_domain(domain);
+  p->set_version(version);
+}
+
+static void ConstructASimpleAddGraph(GraphProto& g, const char* domain) {
+  NodeProto* node = g.add_node();
+  *node->add_input() = "x";
+  *node->add_input() = "y";
+  *node->add_output() = "sum";
+  node->set_op_type("Add");
+  if (domain != nullptr) {
+    node->set_domain(domain);
+  }
+  ValueInfoProto* input1 = g.add_input();
+  input1->set_name("x");
+  SetTypeAndShape(input1->mutable_type()->mutable_tensor_type(), 1, {3, 4, 5});
+  ValueInfoProto* input2 = g.add_input();
+  input2->set_name("y");
+  SetTypeAndShape(input2->mutable_type()->mutable_tensor_type(), 1, {3, 4, 5});
+  ValueInfoProto* output = g.add_output();
+  output->set_name("sum");
+  SetTypeAndShape(output->mutable_type()->mutable_tensor_type(), 1, {3, 4, 5});
+}
+
+TEST_F(GraphTest, SimpleAddWithoutDomain) {
+  std::unique_ptr<ModelProto> m = onnxruntime::make_unique<ModelProto>();
+  m->set_ir_version(3);
+  ImportOpset(*m, "", 10);
+  ConstructASimpleAddGraph(*m->mutable_graph(), nullptr);
+  std::shared_ptr<Model> model;
+  Status st;
+  ASSERT_TRUE((st = Model::Load(std::move(m), model, nullptr, *logger_)).IsOK()) << st.ErrorMessage();
+}
+
+TEST_F(GraphTest, SimpleAddDefaultDomain) {
+  std::unique_ptr<ModelProto> m = onnxruntime::make_unique<ModelProto>();
+  m->set_ir_version(3);
+  ImportOpset(*m, "", 10);
+  ConstructASimpleAddGraph(*m->mutable_graph(), "");
+  std::shared_ptr<Model> model;
+  Status st;
+  ASSERT_TRUE((st = Model::Load(std::move(m), model, nullptr, *logger_)).IsOK()) << st.ErrorMessage();
+}
+
+TEST_F(GraphTest, SimpleAddFutureOpSet) {
+  std::unique_ptr<ModelProto> m = onnxruntime::make_unique<ModelProto>();
+  m->set_ir_version(3);
+  ImportOpset(*m, "", 9999);
+  ConstructASimpleAddGraph(*m->mutable_graph(), "ai.onnx");
+  std::shared_ptr<Model> model;
+  Status st;
+  ASSERT_FALSE((st = Model::Load(std::move(m), model, nullptr, *logger_)).IsOK()) << st.ErrorMessage();
+}
```

Comment:
nit: If the test fails st.ErrorMessage() will be empty.

Reply:
Hi @skottmckay, I split the test cases changes to #3010 .